assets/src/edit-story/components/header/titleMissingDialog.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Plain } from '../button';
+import Dialog from '../dialog';
+import Link from '../link';
+
+const Paragraph = styled.p`
+  font-family: ${({ theme }) => theme.fonts.body1.family};
+  font-size: ${({ theme }) => theme.fonts.body1.size};
+  line-height: ${({ theme }) => theme.fonts.body1.lineHeight};
+  letter-spacing: ${({ theme }) => theme.fonts.body1.letterSpacing};
+`;
+
+const GUIDELINES =
+  'https://amp.dev/documentation/guides-and-tutorials/start/create_successful_stories/#title';
+
+function TitleMissingDialog({ open, onIgnore, onFix }) {
+  return (
+    <Dialog
+      open={open}
+      onClose={onFix}
+      title={__('Missing title', 'web-stories')}
+      actions={
+        <>
+          <Plain onClick={onFix}>{__('Add a title', 'web-stories')}</Plain>
+          <Plain onClick={onIgnore}>
+            {__('Publish without title', 'web-stories')}
+          </Plain>
+        </>
+      }
+    >
+      <Paragraph>
+        {__(
+          'We recommend adding a title to the story prior to publishing',
+          'web-stories'
+        )}
+        {' ('}
+        <Link href={GUIDELINES} target="_blank" rel="noopener noreferrer">
+          {__('learn more', 'web-stories')}
+        </Link>
+        {').'}
```

Comment:
@swissspidy, so this is very LTR centric. What's the proper way to do this? I want to have either one text as

```js
__('We recommend adding a title to the story prior to publishing ([link]learn more[/link]).', 'web-stories')
```

Or two texts as:

```js
__('We recommend adding a title to the story prior to publishing (%s).', 'web-stories')
__('learn more', 'web-stories')
```

But I'm unsure of how to properly do this with React elements to be inserted and still support RTL. Also maybe there languages, that wouldn't use parentheses for something like this (Chinese perhaps?).

I don't think I can use `sprintf`, as I want to insert a full React element. Maybe I could split the string on `%s` and simply add the three elements as child nodes? Would that work in RTL too?

```js
  const [pre, post] = __('We recommend adding a title to the story prior to publishing (%s).', 'web-stories').split('%s');
  ...
      <Paragraph>
        {pre}
        <Link href={GUIDELINES} target="_blank" rel="noopener noreferrer">
          {__('learn more', 'web-stories')}
        </Link>
        {post}
      </Paragraph>
```

The easiest would of course be to skip the parenthesis and just do it as separate sentences:

```js
      <Paragraph>
        {__(
          'We recommend adding a title to the story prior to publishing.',
          'web-stories'
        )}
        <Link href={GUIDELINES} target="_blank" rel="noopener noreferrer">
          {__('Learn more', 'web-stories')}
        </Link>
      </Paragraph>
```

This is less nice to look at but would work correctly when localized, I feel.

Feel free to come up with alternate ideas.

Reply:
@barklund You're essentially running into #1578 here. Until we have a solution for that, I recommend using separate sentences. Still not ideal, but the best there is with the current limitation.

Reply:
Fixed in 18c11d709.